src/sentry/data_secrecy/data_secrecy_logic.py

```diff
@@ -21,19 +19,13 @@ def should_allow_superuser_access(
     else:
         organization = organization_context
 
-    # If organization does not have data-secrecy feature, return True
+    # If organization does not have data-secrecy feature, allow superuser access
     if not features.has("organizations:data-secrecy", organization):
         return True
 
-    # If organization's prevent_superuser_access bitflag is False, return True
+    # If organization's prevent_superuser_access bitflag is False, allow superuser access
     if not organization.flags.prevent_superuser_access:
         return True
 
-    ds = data_secrecy_service.get_data_secrecy_waiver(organization_id=organization.id)
```

Comment:
It's typically safer to remove references to a service in its own PR, then have a second PR to remove the service itself to prevent cross-silo mismatches while things are deploying.

While it's probably ok here (region deploys first, then control), and because your other PR removes the actual usage of this logic, I recommend doing that in the future.